core/src/test/scala/org/apache/spark/FileSuite.scala

```diff
@@ -169,6 +169,27 @@ class FileSuite extends FunSuite with LocalSparkContext {
     assert(output.collect().toList === List(1, 2, 3, 4))
   }
 
+  test("object files compressed") {
+    sc = new SparkContext("local", "test")
+    val normalDir = new File(tempDir, "output_normal").getAbsolutePath
+    val compressedOutputDir = new File(tempDir, "output_compressed").getAbsolutePath
+    val codec = new DefaultCodec()
+    
+    val nums = sc.makeRDD(1 to 4)
+    nums.saveAsObjectFile(normalDir)
+    nums.saveAsObjectFile(compressedOutputDir, classOf[DefaultCodec])
+    
+    // Try reading the output back as an object file
+    val normalFile = new File(normalDir, "part-00000")
+    val normalContent = sc.objectFile[Int](normalDir)
+    assert(normalContent.collect().toList === List(1, 2, 3, 4))
+    
+    val compressedFile = new File(compressedOutputDir, "part-00000" + codec.getDefaultExtension)
```

Comment:
Does this really test if the file is compressed though? It shows it's readable, at least.